src/components/application_manager/src/app_launch/app_launch_ctrl_impl.cc

```diff
@@ -83,6 +83,8 @@ bool HmiLevelSorter(const std::pair<int32_t, ApplicationDataPtr>& lval,
 
 void AppLaunchCtrlImpl::OnDeviceConnected(const std::string& device_mac) {
   SDL_LOG_AUTO_TRACE();
+  sync_primitives::AutoLock lock(launch_ctrl_lock_);
```

Comment:
@ychernysheva what is the purpose of that lock?

Reply:
@AKalinich-Luxoft It seems that the main purpose of this lock is avoiding races, related to free_launchers_ container.